components/pd_client/src/feature_gate.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use semver::{SemVerError, Version};
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+/// The function assumes only major, minor and patch are considered, and they are
+/// all less than u16::MAX, which is 65535.
+const fn ver_to_val(major: u64, minor: u64, patch: u64) -> u64 {
+    major << 32 | minor << 16 | patch
```

Comment:
What if `4.1.0-alpha`? Currently in the master branch PD will give this.

Reply:
Rest LGTM.

Reply:
It will be treated as `4.1.0`. `alpha`, `beta` is ignored on purpose to simplify the problem.